include/CL/cl_exp_defined_builtin_kernels.h

```diff
@@ -261,4 +263,79 @@ typedef struct _cl_dbk_attributes_onnx_inference_exp
   const char **initializer_data;
 } cl_dbk_attributes_onnx_inference_exp;
 
+/**
+ * Collection of different color spaces that can be used
+ * to populate entries in pocl_image_attr_t.
+ */
+enum pocl_color_space_e
+{
+  POCL_COLOR_SPACE_BT601_525,
+  POCL_COLOR_SPACE_BT709,
+  POCL_COLOR_SPACE_DEFAULT = POCL_COLOR_SPACE_BT709,
+};
+
+/**
+ * Collection of different image formats that can be used
+ * to populate entries in pocl_image_attr_t.
+ * @todo describe each type
+ */
+enum pocl_format_e
+{
+  POCL_DF_IMAGE_RGB,
+  POCL_DF_IMAGE_NV12,
+  POCL_DF_IMAGE_NV21,
+  POCL_DF_IMAGE_IYUV,
+};
+
+/**
+ * Collection of image channel ranges that can be used
+ * to populate entries in pocl_image_attr_t.
+ */
+enum pocl_channel_range_e
+{
+  POCL_CHANNEL_RANGE_FULL,
+  POCL_CHANNEL_RANGE_RESTRICTED,
+};
+
+/**
+ * Attributes to describe image data. See \ref pocl_channel_range_e,
+ * \ref pocl_format_e and \ref pocl_color_space_e for valid values.
+ */
+typedef struct
+{
+  cl_int width;
+  cl_int height;
+  cl_int color_space;
+  cl_int channel_range;
+  cl_int format;
+} pocl_image_attr_t;
```

Comment:
Lines 266-311. It looks like reinvention of the cl_image. I don't think we want to go that direction. Perhaps, the OpenCL images should be extended to support the new formats and color spaces?

Reply:
This is likely a good idea in the longer run, but let's bear in mind that DBK is still in experimentation phase. After we have stabilized we should check what are the overlaps with the basic cl_image and merge, but let's not block this for that reason.

Reply:
Fair enough.